assets/src/features/home-page/context/PreviewSettingsContext.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { createContext, useState } from "react";
+import { DEFAULT_AUTOSTART_CAMERA_VALUE, DEFAULT_AUTOSTART_MICROPHONE_VALUE } from "../../../pages/room/consts";
+
+export type PreviewSettings = {
+  cameraAutostart: { status: boolean; setCameraAutostart: (status: boolean) => void };
+  audioAutostart: { status: boolean; setAudioAutostart: (status: boolean) => void };
+};
+
+export const PreviewSettingsContext = createContext<PreviewSettings>({
+  cameraAutostart: { status: false, setCameraAutostart: () => console.log("Error while creating context") },
+  audioAutostart: { status: false, setAudioAutostart: () => console.log("Error while creating context") },
```

Comment:
You could even remove this redundant initialization entirely because you extracted constants (`DEFAULT_AUTOSTART_CAMERA_VALUE` and `DEFAULT_AUTOSTART_MICROPHONE_VALUE`)

```ts
createContext<PreviewSettings | undefined>()

export const usePreviewSettings = () => {
    const context = useContext(PreviewSettingsContext) 
    if(!context) throw "Error while creating PreviewSettingsContext context" 
    // or "usePreviewSettings must be used within a PreviewSettingsProvider"
    return context;
}

...


```